src/ssl.c

```diff
@@ -16570,9 +16574,23 @@ WOLFSSL_BIGNUM *wolfSSL_ASN1_INTEGER_to_BN(const WOLFSSL_ASN1_INTEGER *ai,
         return NULL;
     }
 
-    if (SetIndividualExternal(&bn, &mpi) != SSL_SUCCESS) {
-        return NULL;
+    /* SetIndividualExternal mallocs bn in the case that bn is NULL
+     * mp_clear needs called because mpi is copied and causes memory leak with
+     * --disable-fastmath */
+    if (bn == NULL) {
+        if (SetIndividualExternal(&bn, &mpi) != SSL_SUCCESS) {
+            wolfSSL_BN_free(bn);
```

Comment:
Shouldn't SetIndividualExternal() free the bn it allocs in failure case instead?  And shouldn't this function call mp_clear() once to simplify the branches?